lightning/src/blinded_path/message.rs

```diff
@@ -74,16 +68,106 @@ impl Writeable for ForwardTlvs {
 	}
 }
 
+/// Similar to [`ForwardTlvs`], but these TLVs are for the final node.
+pub(crate) struct ReceiveTlvs {
+	pub tlvs: Option<RecipientData>
+}
+
+impl ReceiveTlvs {
+	pub fn new(tlvs: Option<RecipientData>) -> Self {
+		ReceiveTlvs { tlvs }
+	}
+}
+
+/// Represents additional data appended along with the sent reply path.
+///
+/// This data can be utilized by the final recipient for further processing
+/// upon receiving it back.
+#[derive(Clone, Debug)]
+pub enum RecipientData {
+	/// Represents the data specific to [`OffersMessage`]
+	///
+	/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+	OffersContext(OffersData),
+
+	/// Represents the data appended with Custom Onion Message.
+	CustomContext(Vec<u8>),
+}
+
+/// Contains the data specific to [`OffersMessage`]
+///
+/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+#[derive(Clone, Debug)]
+pub struct OffersData {
+	/// Payment ID of the outbound BOLT12 payment.
+	pub payment_id: Option<PaymentId>
+}
+
+impl RecipientData {
+	/// Creates a new RecipientData::OffersContext instance.
+	pub fn new_for_offers(payment_id: Option<PaymentId>) -> Self {
+		RecipientData::OffersContext(OffersData {
+			payment_id,
+		})
+	}
+
+	/// Creates a new RecipientData::CustomContext instance.
+	pub fn new_for_custom(data: Vec<u8>) -> Self {
+		RecipientData::CustomContext(data)
+	}
```

Comment:
We can probably drop this as well.

Reply:
Done! [8e4e1aa](https://github.com/lightningdevkit/rust-lightning/commit/8e4e1aabadb5fda9f4d0ec2c845263c94c273643)
Thanks!